ppx/ast_external_process.ml

```diff
@@ -826,16 +826,31 @@ let external_desc_of_non_obj (loc : Location.t) (st : external_desc)
       (* PR #2162 - since when we assemble arguments the first argument in
          [@@send] is ignored
       *)
-      match arg_type_specs with
-      | [] ->
+      match (arg_type_specs, new_name) with
+      | [], _ ->
           Location.raise_errorf ~loc
             "Ill defined attribute %@send(the external needs to be a regular \
              function call with at least one argument)"
-      | { arg_type = Arg_cst _; arg_label = _ } :: _ ->
+      | { arg_type = Arg_cst _; arg_label = _ } :: _, _ ->
           Location.raise_errorf ~loc
             "Ill defined attribute %@send(first argument can't be const)"
-      | _ :: _ ->
-          Js_send { splice; name; js_send_scopes = scopes; pipe = false })
+      | _, `Nm_payload _ ->
+          Location.raise_errorf ~loc
+            "Incorrect FFI attribute found: (%@new should not carry a payload \
+             here)"
+      | _ :: _, `Nm_na ->
+          Js_send
+            {
+              splice;
+              name;
+              js_send_scopes = scopes;
+              pipe = false;
+              new_ = false;
+            }
+      | _ :: _, (`Nm_val _ | `Nm_external _) ->
```

Comment:
`Nm_external` can't be produced with "new", right?

Reply:
`Nm_payload` can't, but `Nm_external` can.